backend/src/preprocessing.py

```diff
@@ -148,25 +151,39 @@ def registration(packet):
     [in] packet : pipeline packet
     [post] images registered in place
     """
-    return  # TODO remove
-    # TODO how are these gonna be and what to return
-    reference_color, align_color = packet
+    subject = packet.get_subject()
+    align_color = subject[1]
+    reference_color = subject[0]
```

Comment:
I think these are making copies. It would be better to just call subject[n] when you need one. For clarity you could make a constant in place of "n"